kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskCreator.java

```diff
@@ -226,6 +221,43 @@ public long determineStartingOffset(long currentHeadOffsetForPartition) {
     }
   }
 
+  /**
+   * Currently, a recovery task will index all the messages into a single chunk. So, if we create a
+   * recovery task with a large offset range, it will result in large chunks. So, we create multiple
+   * recovery tasks where each task indexes maxMessagesPerRecoveryTask worth of messages. This keeps
+   * all the recovery task chunk sizes roughly the same size.
+   */
+  @VisibleForTesting
+  void createRecoveryTasks(
+      final String partitionId,
+      final long startOffset,
+      final long endOffset,
+      final long maxMessagesPerRecoveryTask) {
+    long beginingOffset = startOffset;
+    // Create tasks until there are no messages left to be indexed. Offsets are inclusive so adding
+    // +1 to count messages to be indexed.
+    while ((endOffset - beginingOffset + 1) > 0) {
+      createRecoveryTask(
+          partitionId,
+          beginingOffset,
+          Math.min(beginingOffset + maxMessagesPerRecoveryTask - 1, endOffset));
+      beginingOffset = beginingOffset + maxMessagesPerRecoveryTask;
+    }
+  }
+
+  private void createRecoveryTask(String partitionId, long startOffset, long endOffset) {
+    final long creationTimeEpochMs = Instant.now().toEpochMilli();
+    final String recoveryTaskName = getRecoveryTaskName(creationTimeEpochMs, partitionId);
+    recoveryTaskMetadataStore.createSync(
+        new RecoveryTaskMetadata(
+            recoveryTaskName, partitionId, startOffset, endOffset, creationTimeEpochMs));
+    LOG.info(
+        "Created recovery task {} to catchup. Moving the starting offset to head at {}",
+        recoveryTaskName,
+        endOffset);
+    recoveryTasksCreated.increment();
```

Comment:
It would be beneficial to also include a tag for the partition ID on the counter as well.